src/System.Memory/src/System/SpanHelpers.T.cs

```diff
@@ -182,5 +182,74 @@ public static bool SequenceEqual<T>(ref T first, ref T second, int length)
         NotEqual: // Workaround for https://github.com/dotnet/coreclr/issues/9692
             return false;
         }
+
+        public static unsafe bool StartsWith<T>(ref T searchSpace, int searchSpaceLength, ref T value, int valueLength)
```

Comment:
BTW: The searchSpaceLength argument is useless because of it is just used for debug assert.